### PR TITLE
feat: search ignore diacritical marks

### LINKED ISSUE #1405: search accents marks
## Feature request
It posible that search plugin ignore accents marks (tildes) ?
  
#### What problem does this feature solve?
id like that the words with accents to be the same with out accents.

#### What does the proposed API look like?
In the Spanish lenguage is important write accents, but in the search, the peoples does not usually add them.

#### How should this be implemented in your opinion?
adding some kind of routine that ignores accents in vowels: á é í ó ú
for example: **artículo** or **articulo** to be the same

#### Are you willing to work on this yourself?
yes

### PR DESCRIPTION
**Pull request by sy-records**
**Summary**

search ignore diacritical marks

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Repo settings
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

resolve #1405

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [x] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

### CODE REVIEW

**Comment by sy-records** [comment]
Unfortunately, normalize does not support IE.

**Comment by jhildenbiddle** [comment]
@sy-records --

Given the near-universal support of `normalize` in other browsers, I think we can just mention the need for a ponyfill to support IE11 and provide a URL (below):

* https://polyfill.io/v3/polyfill.min.js?features=String.prototype.normalize

We should verify that this ponyfill does in fact work, and specify that it needs to be included *before* any docsify-related `<script>` elements.

**Comment by sy-records** [comment]
I don't have IE11 to testing 😥

If we use this, do we want all users who need to search to add this `polyfill`?

**Comment by jhildenbiddle** [comment]
I can test in IE11 once the PR is ready. 

As for the ponyfill, it should be optional. Docsify should feature check for `normalize` support and use it if available. If not, then the search plugin should function as it always has. The docs should suggest the ponyfill only for those concerned with diacritical marks and IE11 support. This way it's a non- breaking enhancement.

**Comment by sy-records** [comment]
@jhildenbiddle done

**Review by jhildenbiddle — CHANGES_REQUESTED** [review]
Verified working in modern browsers. Also verified working with and without polyfill in IE11. Great!

I did notice a few issues:

1. Search results are not displayed with their original diacritical marks. Notice in the screenshot below that the page contains the string "Let's go to café today". A search for "cafe" (no diacritical marks) matches "café" as it should, but the search result displayed under the search field shows the normalized version ("cafe") instead of the original text ("café").

   Screen Shot 2020-12-15 at 1 42 11 AM

1. Notice in the screenshot below that searching for "let" displays the matching search result text as "...Let's go to a cafe t...", but searching for "cafe" displays "...<span>#</span>39;s go to a cafe t...". That  "<span>#</span>39;" looks like an incorrectly rendered HTML entity, but I haven't looked at the code to verify.
   
   Screen Shot 2020-12-15 at 1 51 33 AM
   Screen Shot 2020-12-15 at 1 51 15 AM

1. It looks like apostrophes break the search result highlighting.

   Screen Shot 2020-12-15 at 1 52 36 AM

**FYI:** Docsify is currently broken in IE11 due to issues with marked and the carbon ads plugin. I've already submitted fixes for marked (https://github.com/markedjs/marked/pull/1877) and will address carbon ads soon. My tests in IE11 were done with a patched version of marked.

**Comment by sy-records** [comment]
@jhildenbiddle Thanks for the test.

The third one is solved.

The second is an escapeHtml, I don t know how to deal with it for the time being.

The first question, I need to think about how to modify... 😂

File: test/e2e/search.test.js [hunk 1]
```diff
@@ -105,4 +105,23 @@ describe('Search Plugin Tests', function() {
...
+          # Qué es
+
+          docsify genera su sitio web de documentación sobre la marcha. A diferencia de GitBook, no genera archivos estáticos html. En cambio, carga y analiza de forma inteligente sus archivos de Markdown y los muestra como sitio web. Todo lo que necesita hacer es crear un index.html para comenzar y desplegarlo en GitHub Pages.
+        `,
+      },
+      scriptURLs: ['/lib/plugins/search.min.js'],
+    };
+    await docsifyInit(docsifyInitConfig);
+    await page.fill('input[type=search]', 'documentacion');
+    await expect(page).toEqualText('.results-panel h2', 'Que es');
```
Comment (anikethsaha) [inline test/e2e/search.test.js:122]: is it possible to show the accent in the result ?

sorry if I missed any previous discussion regarding this. LMK

**Review by anikethsaha — CHANGES_REQUESTED** [review]
one small suggestion. 
Otherwise great stuff. 

THanks @sy-records

File: test/e2e/search.test.js (hunk 1 above)
Comment (sy-records) [reply test/e2e/search.test.js:122]: I don't have a good idea how to do it...

Comment (jhildenbiddle) [reply test/e2e/search.test.js:122]: Looks like @sy-records got 2 of the 3 I listed. Perhaps we should open a new issue regarding missing diacritical marks in the search results so we can merge and release this PR.

**Review by jhildenbiddle — APPROVED** [review]
LGTM

File: test/e2e/search.test.js (hunk 1 above)
Comment (sy-records) [reply test/e2e/search.test.js:122]: done. #1491

Comment (jhildenbiddle) [reply test/e2e/search.test.js:122]: Created a new issue for missing diacritical remarks in search results (#1492).

**Pull request merged**